### ISSUE TITLE
_ in twitter handle causes error with url linking to account

### ISSUE THREAD

**Issue by MichelineCampbell** [mitchelloharawild/vitae#117]
including an underscore in the twitter: entry for vitae YAML resulted in URL breaking with "/protect%20_" (e.g. my handle was @michcampbell_ - redirected to https://twitter.com/michcampbell/protect%20_ (page that does not exist).)

**Comment by mitchelloharawild** [comment]
Which template are you using?

**Comment by MichelineCampbell** [comment]
awesomecv

**Comment by mitchelloharawild** [comment]
I'm unable to reproduce this issue. The URL seems to take me to the right place, although I also can't access your twitter account (even following the crawled URL via google).

````r
---
name: Marie
surname: Curie
position: "Professor"
address: "School of Physics & Chemistry, École Normale Supérieure"
phone: +1 22 3333 4444
www: mariecurie.com
email: "Marie.Curie@ens.fr"
twitter: michcampbell_
github: michcampbell_
linkedin: michcampbell_
date: "`r format(Sys.time(), '%B %Y')`"
output: vitae::awesomecv
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, warning = FALSE, message = FALSE)
library(vitae)
library(dplyr)
```

# Some stuff about me

* I poisoned myself doing research.
* I was the first woman to win a Nobel prize
* I was the first person and only woman to win a Nobel prize in two different sciences.

# Education

```{r}
library(tibble)
tribble(
  ~ Degree, ~ Year, ~ Institution, ~ Where,
  "Informal studies", "1889-91", "Flying University", "Warsaw, Poland",
  "Master of Physics", "1893", "Sorbonne Université", "Paris, France",
  "Master of Mathematics", "1894", "Sorbonne Université", "Paris, France"
) %>%
  detailed_entries(Degree, Year, Institution, Where)
```

# Nobel Prizes

```{r}
tribble(
  ~Year, ~Type, ~Desc,
  1903, "Physics", "Awarded for her work on radioactivity with Pierre Curie and Henri Becquerel",
  1911, "Chemistry", "Awarded for the discovery of radium and polonium"
) %>%
  brief_entries(
    glue::glue("Nobel Prize in {Type}"),
    Year,
    Desc
  )
```

# Publications

```{r}
scholar::get_publications("EmD_lTEAAAAJ&EmD_lTEAAAAJ&") %>%
  filter(cites > 30) %>%
  distinct(title, .keep_all = TRUE) %>%
  transmute(bibtype = "Article", author = as.character(author),
            title = as.character(title),
            journaltitle = as.character(journal), year, key = row_number()) %>%
  RefManageR::as.BibEntry() %>%
  RefManageR::WriteBib("curie.bib")

bibliography_entries("curie.bib")
```

````

[test-twitter.pdf](https://github.com/mitchelloharawild/vitae/files/4526633/test-twitter.pdf)

**Comment by MichelineCampbell** [comment]
Yep, I changed my twitter handle as a quick personal fix last night, so michampbell_ won't take you there anyway.

**Comment by mitchelloharawild** [comment]
Perhaps it is a PDF reader issue? Could you try to follow the link in the pdf provided above?

**Comment by MichelineCampbell** [comment]
Possibly - I'll try on my home computer this evening, but the link worked fine just then in both chrome's PDF viewer and adobe acrobat at work. Sorry, I didn't think to check by PDF reader initially (although links to github and email worked fine, so I think it was still the underscore causing a problem).

**Comment by MichelineCampbell** [comment]
OK, I just realised I'm a numpty and I had recompiled it in moderncv.

**Comment by MichelineCampbell** [comment]
OK I just realised I'm a numpty and had recompiled it in moderncv. Sorry for the mess around.

---
name: Dr. Micheline Campbell
date: "`r format(Sys.time(), '%B, %Y')`"
qualifications: BA(Hons), PhD
position: Scientist, QLD Dept. of Natural Resources, Mines, and Energy
address: Rockhampton, Australia
phone: +61 411 282 657
email: michelineleecampbell@gmail.com
github: MichelineCampbell
twitter: michcampbell_
headcolor: 414141
output: 
  vitae::moderncv:
    theme: classic
    keep_tex: true
---
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, warning = FALSE, message = FALSE)
library(vitae)
library(scholar)
library(readr)
library(tidyverse)
library(glue)
library(lubridate)
```

# Professional Summary

**Comment by mitchelloharawild** [comment]
Thanks for the reprex, I'll get this fixed before the next release.

**Comment by mitchelloharawild** [comment]
Hey @robjhyndman, is there some macro which can remove escapes from text (or a better approach entirely).

The issue is that pandoc escapes `twitter_` into `twitter\_`, which is then passed into `\href{}` which doesn't require escaping `_` in the url.

**Comment by robjhyndman** [comment]
See https://tex.stackexchange.com/a/42332/143 for a discussion on creating
a macro to remove a backslash.

Here is the macro.

\newcommand{\removeabs}[1]{%
  \ifcat\relax\noexpand#1%
    \expandafter\expandafter\expandafter\@gobble\expandafter\string
  \fi
  #1%
}

I think you should be able to use \href{\removeabs{$twitter$}}.

  _________________________
*Rob J Hyndman*
robjhyndman.com

**Comment by mitchelloharawild** [comment]
Doesn't look like it worked. An alternate solution is to use `\textunderscore`, but this isn't ideal.

In the csl I've used `\href{\removeabs{https://www.twitter.com/#3}}{#3}`.

**Comment by mitchelloharawild** [comment]
Closing as this bug is specific to the `moderncv` template, which should be fixed in the `moderncv` repository. This bug is described here: https://bugs.launchpad.net/moderncv/+bug/1330710

This will likely be fixed as a side-effect of #69, where the template specific social link support will be replaced with a custom `{vitae}` solution that uses https://github.com/mitchelloharawild/icon/ to provide any social icon.

**Comment by mitchelloharawild** [comment]
Re-opening as I have found a PR to moderncv which fixes this issue: https://github.com/xdanaux/moderncv/pull/77
